models/src/node/drain.rs

```diff
@@ -288,13 +288,18 @@ async fn wait_for_deletion(k8s_client: &kube::Client, pod: &Pod) -> Result<(), e
                 event!(Level::INFO, "Pod {} deleted.", pod.name(),);
                 break;
             }
-            Ok(_) => {
-                event!(
-                    Level::DEBUG,
-                    "Pod '{}' not yet deleted. Waiting {}s.",
-                    pod.name(),
-                    DELETION_CHECK_INTERVAL.as_secs_f64()
-                );
+            Ok(p) => {
+                if p.uid() != pod.uid() {
+                    event!(Level::INFO, "Pod {} deleted.", p.name(),);
```

Comment:
nit:

I'd suggest using "match guards" here to keep the indentation level as before. e.g.

```rust
Err(kube::Error::Api(e)) if e.code == 404 => {...},
Ok(p) if p.uid() != pod.uid() => {...}
Ok(p) => {...},
```